pwiz_tools/Skyline/TestUtil/AbstractUnitTestEx.cs

```diff
@@ -178,5 +181,331 @@ public string GetSystemResourceString(string resourceId, params object[] args)
+            Assert.IsNotNull(getProjectDirectory, "GetProjectDirectory function cannot be null when loading HTTP interactions");
+
+            var jsonPath = GetHttpRecordingFilePathFromType(testType, getProjectDirectory);
+
+            // CONSIDER: This is currently required for recording from the web when the JSON file does not yet exist.
+            // It might be nice to be able to fail here when the file is really necessary, since it would allow us to
+            // have the error message include the expected file path.
+            if (!File.Exists(jsonPath))
+                return new List<HttpInteraction>();
+
+            var json = File.ReadAllText(jsonPath, Encoding.UTF8);
+            var data = JsonConvert.DeserializeObject<HttpRecordingData>(json);
+            var interactions = data?.HttpInteractions ?? new List<HttpInteraction>();
+            
+            // Resolve response body references after deserialization
+            ResolveResponseBodyReferences(interactions);
+            
+            return interactions;
+        }
+
+        /// <summary>
+        /// Resolves response body references by copying content from referenced indices.
+        /// </summary>
+        private static void ResolveResponseBodyReferences(List<HttpInteraction> interactions)
+        {
+            if (interactions == null || interactions.Count == 0)
+                return;
+
+            for (int i = 0; i < interactions.Count; i++)
+            {
+                var interaction = interactions[i];
+                if (interaction.ResponseBodyIndex.HasValue)
+                {
+                    int referencedIndex = interaction.ResponseBodyIndex.Value;
+                    
+                    // Validate index is within bounds
+                    if (referencedIndex < 0 || referencedIndex >= interactions.Count)
+                    {
+                        throw new InvalidOperationException(
+                            $"Invalid ResponseBodyIndex {referencedIndex} in interaction {i}. " +
+                            $"Index must be between 0 and {interactions.Count - 1}.");
+                    }
+
+                    // Validate we're not creating a circular reference
+                    if (referencedIndex == i)
+                    {
+                        throw new InvalidOperationException(
+                            $"Circular reference detected: interaction {i} references itself.");
+                    }
+
+                    var referencedInteraction = interactions[referencedIndex];
+                    
+                    // Copy response body content from the referenced interaction
+                    interaction.ResponseBody = referencedInteraction.ResponseBody;
+                    interaction.ResponseBodyIsBase64 = referencedInteraction.ResponseBodyIsBase64;
+                    interaction.ResponseBodyLines = referencedInteraction.ResponseBodyLines != null

```

Comment:
You could do:
`interaction.ResponseBodyLines = referencedInteraction.ResponseBodyLines?.ToList();`